app/lib/features/pins/pages/pin_details_page.dart

```diff
@@ -75,49 +76,83 @@ class _PinDetailsPageState extends ConsumerState<PinDetailsPage> {
           ),
         );
       },
-      loading: () => Skeletonizer(
-        child: Text(L10n.of(context).loadingPin),
-      ),
-      error: (e, s) {
-        _log.severe('Error loading pin', e, s);
-        return Text(
-          L10n.of(context).errorLoadingPin(e),
+      loading: _contentLoader,
+      error: (error, stack) {
+        _log.severe('Error loading pin', error, stack);
+        return ErrorPage(
+          background: _contentLoader(),
+          error: error,
+          stack: stack,
+          onRetryTap: () {
+            ref.invalidate(pinProvider(widget.pinId));
+          },
         );
       },
     );
   }
 
-  // pin actions menu builder
-  Widget _buildActionMenu() {
-    final pinData = ref.watch(pinProvider(widget.pinId));
-    return pinData.when(
-      data: (pin) {
-        //Get my membership details
-        final membership =
-            ref.watch(roomMembershipProvider(pin.roomIdStr())).valueOrNull;
-        final canRedactData = ref.watch(canRedactProvider(pin));
-        bool canPost = membership?.canString('CanPostPin') == true;
-        bool canRedact = canRedactData.valueOrNull == true;
-        return PopupMenuButton<String>(
-          key: PinDetailsPage.actionMenuKey,
-          icon: const Icon(Atlas.dots_vertical_thin),
-          itemBuilder: (context) =>
-              _buildAppBarActionMenuItems(pin, canPost, canRedact),
-        );
-      },
-      loading: () => Skeletonizer(child: Text(L10n.of(context).loadingPin)),
-      error: (e, s) {
-        _log.severe('Error loading pin', e, s);
-        return const SizedBox.shrink();
-      },
+  Widget _contentLoader() {
+    // not the nicest loading animation, but
+    return SingleChildScrollView(
+      child: Column(
+        children: [
+          const SizedBox(height: 20),
+          _loadingPinHeaderUI(),
+          const SizedBox(height: 20),
+          AttachmentSectionWidget.loading(),
+          const SizedBox(height: 20),
+          CommentsSection.loading(context),
+        ],
+      ),
+    );
+  }
```

Comment:
Why not making separate `Skeletonizer` and using it for both loading and error dialog background?

Reply:
but that is what this does...
```

      loading: _contentLoader,
      error: (error, stack) {
        _log.severe('Error loading pin', error, stack);
        return ErrorPage(
          background: _contentLoader(),
          error: error,
          stack: stack,
          onRetryTap: () {
            ref.invalidate(pinProvider(widget.pinId));
          },
        );
 ```